### PR TITLE
Update Fastlane to upload Simplenote Internal to AppCenter

### PR DESCRIPTION
**Pull request by loremattei**
This PR updates the Fastlane configuration to switch binary upload from HockeyApp to MSAppCenter because of HockeyApp sunsetting. 

### Test
1. Create a new [API Token in AppCenter](https://appcenter.ms/settings/apitokens).
2. Add a `APPCENTER_API_TOKEN` entry to your `~/.simplenoteios-env.default` with the token above. 
3. Bump the version of the app.
4. `bundle install`.
5. Run `bundle exec fastlane build_and_upload_internal` and verify that the app is built and uploaded to App Center.

### Review
 Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

**Comment by jkmassel** [comment]
In order to fully test this PR, I think I need to be added to the `Hogwarts-Organization/Simplenote` project in App Center. 

But so far the rest works!

**Comment by loremattei** [comment]
@jkmassel thanks for looking!
You should be added to the project now, though you may need to login on the web the first time you try to access App Center.
Let me know if you have other issues with this!

File: fastlane/Fastfile [hunk 1]
```diff
@@ -191,13 +191,13 @@ ENV["PROJECT_ROOT_FOLDER"]="./"
...
-        notify: "0",
-        status: "1",
-        ipa: "./build/Simplenote Internal.ipa",
-        dsym: "./build/Simplenote.app.dSYM.zip",
+    appcenter_upload(
+      api_token: ENV["APPCENTER_API_TOKEN"],
+      owner_name: "Hogwarts-Organization",
+      owner_type: "organization", 
+      app_name: "Simplenote",
+      file: "./build/Simplenote\ Internal.ipa",
```
Comment (jkmassel) [inline fastlane/Fastfile]: Me again! 🤦‍♂ 

When running this, I got an error saying "Couldn't find build file at path '/Users/jeremymassel/Projects/simplenote-ios/build/Simplenote.ipa'".

It looks like by default, `appcenter_upload` might use the value of `IPA_OUTPUT_PATH`, which in this case doesn't resolve to the new path after the `mv` command above.

In the [example Fastfile](https://github.com/microsoft/fastlane-plugin-appcenter/blob/master/fastlane/Fastfile) for appcenter, I found that they use `file` in just one case, but `ipa` everywhere else  – changing this key to `ipa` made the upload work for me, so I'm wondering if that is "more" correct?

WDYT?

Comment (loremattei) [reply fastlane/Fastfile]: mmm... looking at the code `ipa` seems to be deprecated[ here](https://github.com/microsoft/fastlane-plugin-appcenter/blob/0d66522efafad3849e177b475477d30cc5c923e3/lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb#L383), while `file` seems to be the [current way to go ](https://github.com/microsoft/fastlane-plugin-appcenter/blob/0d66522efafad3849e177b475477d30cc5c923e3/lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb#L399).

It seems it performs more checks with the `file` option... let me check if some of them may fail for us for some reason.

**Comment by loremattei** [comment]
Thanks for the review!

**Pull request merged**